### PR TITLE
#4622 [Manager] Reset the from data when successful topic creation

### PR DESCRIPTION
**Pull request by eechih**
### Checklist
- [x] [Review Contribution Guidelines](https://ohara.readthedocs.io/en/latest/contributing.html)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#ohara-team))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #4622 )
- [ ] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-manager/client/src/components/Workspace/Settings/pages/TopicPage/SharedTopicTable.js [hunk 1]
```diff
@@ -29,7 +29,22 @@ function SharedTopicTable() {
   const { close: closeSettingsDialog } = context.useEditWorkspaceDialog();
 
   const handleCreate = topicToCreate => {
-    createAndStartTopic(topicToCreate);
+    return new Promise((resolve, reject) => {
```
Comment (konekoya) [inline ohara-manager/client/src/components/Workspace/Settings/pages/TopicPage/SharedTopicTable.js:32]: Is this the way we will be using when it comes to handling an epic's success or failure from components?

File: ohara-manager/client/src/store/epics/topic/createAndStartTopicEpic.js [hunk 2]
```diff
@@ -46,13 +46,15 @@ export default action$ =>
...
           }
+          if (options?.onSuccess) options.onSuccess();
         }),
         catchError(res => {
           if (paperApi) {
             paperApi.updateElement(params.id, {
               status: CELL_STATUS.failed,
             });
           }
+          if (options?.onFailure) options.onFailure();
```
Comment (saivirtue) [inline ohara-manager/client/src/store/epics/topic/createAndStartTopicEpic.js:57]: should be `options.onFailure(res)`

Comment (eechih) [reply ohara-manager/client/src/store/epics/topic/createAndStartTopicEpic.js:57]: Yes, I will update

File: ohara-manager/client/src/components/Workspace/Settings/pages/TopicPage/SharedTopicTable.js (hunk 1 above)
Comment (eechih) [reply ohara-manager/client/src/components/Workspace/Settings/pages/TopicPage/SharedTopicTable.js:32]: return Promise 是為了讓 final-form 知道 submit 成功或失敗的時機

Comment (eechih) [reply ohara-manager/client/src/components/Workspace/Settings/pages/TopicPage/SharedTopicTable.js:32]: Final-Form onSubmit 支援 Asynchronous with a Promise。我目前只想到這種方式，讓 form submit 成功後清除 form data。

File: ohara-manager/client/src/store/epics/topic/createAndStartTopicEpic.js (hunk 2 above)
Comment (konekoya) [reply ohara-manager/client/src/store/epics/topic/createAndStartTopicEpic.js]: The variable name should also be renamed to error :)

File: ohara-manager/client/src/components/Workspace/Settings/pages/TopicPage/SharedTopicTable.js (hunk 1 above)
Comment (konekoya) [reply ohara-manager/client/src/components/Workspace/Settings/pages/TopicPage/SharedTopicTable.js:32]: Okay, just wonder if this is the way that we will be using :)

File: ohara-manager/client/src/store/epics/topic/createAndStartTopicEpic.js (hunk 2 above)
Comment (eechih) [reply ohara-manager/client/src/store/epics/topic/createAndStartTopicEpic.js]: > The variable name should also be renamed to error

@konekoya No problem

**Pull request merged**